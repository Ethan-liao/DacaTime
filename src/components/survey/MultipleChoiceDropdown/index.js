import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import QuestionContainer from '../QuestionContainer';
import { updateField } from '../../../actions';

const changeHandler = dispatch => targetKeyPath => event => {
  const value = event.target.value;
  dispatch(updateField(targetKeyPath, value));
};

const MultipleChoiceDropdown = props => {
  const updateKeyPath = [props.name];
  const value = props.value || '';

  const fields = props.fields.map(({ label, name }) =>
    <option key={name} value={name}>
      {label}
    </option>,
  );

  return (
    <QuestionContainer {...props}>
      <select
        className="survey-select"
        onChange={changeHandler(props.dispatch)(updateKeyPath)}
        value={value}
      >
        <option key={''} value="">
          Select a choice...
        </option>
        {fields}
      </select>
    </QuestionContainer>
  );
};
MultipleChoiceDropdown.propTypes = {
  questionId: PropTypes.number.isRequired,
  prompt: PropTypes.string.isRequired,
  explanation: PropTypes.string,
  fields: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  ).isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  dispatch,
});

const mapStateToProps = (state, ownProps) => ({
  value: state.form[ownProps.name],
  ...ownProps,
});

export const component = MultipleChoiceDropdown;
export default connect(mapStateToProps, mapDispatchToProps)(component);
