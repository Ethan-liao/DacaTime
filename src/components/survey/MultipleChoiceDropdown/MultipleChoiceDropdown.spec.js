import React from 'react';
import MultipleChoiceDropdown from './index';
import { dropDownQuestion } from '../../../util/testData';
import reducer from '../../../reducers/survey';
import { setup } from '../../../util/testUtils';
import renderer from 'react-test-renderer';

const { component } = setup(MultipleChoiceDropdown, reducer, dropDownQuestion);

it('renders without crashing', () => {
  renderer.create(component);
});

it('renders correctly', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
