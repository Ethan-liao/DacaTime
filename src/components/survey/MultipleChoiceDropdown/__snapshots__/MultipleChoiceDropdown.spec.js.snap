exports[`test renders correctly 1`] = `
<div
  className="dt-panel dt-question-container">
  <div
    className="dt-question-header">
    What is your current marital status?
  </div>
  <select
    className="survey-select"
    onChange={[Function]}
    value="">
    <option
      value="">
      Select a choice...
    </option>
    <option
      value="married">
      Married
    </option>
    <option
      value="widowed">
      Widowed
    </option>
    <option
      value="single">
      Single
    </option>
    <option
      value="divorced">
      Divorced
    </option>
  </select>
</div>
`;
